### PR TITLE
Use unionId instead of openId

### PR DESCRIPTION
**Pull request by ChrisBr**
Unlike the openId the unionId is stable accross applications.

As often one application has several WeChat applications e.g. for Android, iOS and Web
the unionId should get used otherwise a user which signed up on e.g. Android can not signup
with the same WeChat account on Web or iOS as it would have a different openID for each
of these applications.

> OpenID is an unique encrypted WeChat ID for each user of an OA (including Mini Program).

> UnionID is an unique ID to identify a single user under the same WeChat open platform account.
If a developer possesses multiple mobile applications, website applications, and OAs (including Mini Programs),
they can use a UnionID to differentiate the uniqueness of users.
In other words, UnionID offers a mechanism for developers to acquire a user’s data from multiple
platforms and attribute actions back to the specific user.

https://wiredcraft.com/blog/wechat-for-dummies-user-data-and-authorization/
https://stackoverflow.com/questions/40044572/wechat-open-platform-openid-and-unionid

This change would break existing users of the strategy as we change the `uid` attribute.
However, I don't think there is a way to make this backward compatible :thinking:.

### CODE REVIEW

**Comment by ChrisBr** [comment]
Not sure what is going on with the specs in Travis, they work locally for me

```ruby
rake aborted!
NoMethodError: undefined method `last_comment' for #<Rake::Application:0x0000000002201f78>
/home/travis/.rvm/gems/ruby-2.3.7/gems/rspec-core-2.99.2/lib/rspec/core/rake_task.rb:143:in `initialize'
/home/travis/build/NeverMin/omniauth-wechat-oauth2/Rakefile:5:in `new'
/home/travis/build/NeverMin/omniauth-wechat-oauth2/Rakefile:5:in `<top (required)>'
/home/travis/.rvm/gems/ruby-2.3.7/gems/rake-13.0.1/exe/rake:27:in `<top (required)>'
/home/travis/.rvm/gems/ruby-2.3.7/bin/ruby_executable_hooks:15:in `eval'
/home/travis/.rvm/gems/ruby-2.3.7/bin/ruby_executable_hooks:15:in `<main>'
(See full trace by running task with --trace)
```

**Comment by Eric-Guo** [comment]
`last_comment` is removed at rake v12, so need binding rake version to v11 to resolve this problem.

**Comment by ChrisBr** [comment]
> last_comment is removed at rake v12, so need binding rake version to v11 to resolve this problem.


@Eric-Guo I see, I reverted 91731a5.

**Comment by Eric-Guo** [comment]
Just release 0.1.3 and contain the #27 bug fix, kindly rebase to master for this PR.

**Comment by Eric-Guo** [comment]
>  I don't think there is a way to make this backward compatible

After I merge #21 seems there is a backward compatible to do things, fetch `uid` like below will using current way.

```ruby
  def self.from_omniauth(auth)
    instructor = where(provider: auth.provider, uid: auth.uid).first_or_initialize do |instructor|
      instructor.nickname = auth.info.nickname
      instructor.sex = auth.info.sex
      instructor.city = auth.info.city
      instructor.district = auth.info.province
    end
    instructor.nickname ||= auth.info.nickname
    instructor.sex ||= auth.info.sex
    instructor.city ||= auth.info.city
    instructor.district ||= auth.info.province
    instructor
  end
```

Change `uid` to `unionid` will meet your requirement in this PR.

```diff
-    instructor = where(provider: auth.provider, uid: auth.uid).first_or_initialize do |instructor|
+    instructor = where(provider: auth.provider, unionid: auth.unionid).first_or_initialize do |instructor|
```

So I guess this PR is no need?

**Comment by ChrisBr** [comment]
> So I guess this PR is no need?

Yeah, in regards to backward compatibility it makes sense to keep it in the application logic. Thanks 👍

**Comment by Eric-Guo** [comment]
Thanks, v0.2.0 release to contain #21 content.
